addons/mail/static/src/models/link_preview.js

```diff
@@ -0,0 +1,104 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, many, one } from '@mail/model/model_field';
+
+registerModel({
+    name: 'LinkPreview',
+    identifyingFields: ['id'],
+    recordMethods: {
+        async remove() {
+            await this.messaging.rpc({
+                route: '/mail/link_preview/delete',
+                params: { link_preview_id: this.id },
+            }, { shadow: true });
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsDeletable() {
+            if (!this.messaging) {
+                return false;
+            }
+            return this.message.canBeDeleted ||
+                (this.message.author && this.message.author === this.messaging.currentPartner) ||
+                (this.message.guestAuthor && this.message.guestAuthor === this.messaging.currentGuest);
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsGif() {
+            return this.mimetype === 'image/gif';
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsImage() {
+            return this.type === 'image' && !this.isGif;
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsVideo() {
+            const type = this.type.split('.');
+            return type.includes('video') && !this.isGif;
```

Comment:
is there a type that includes video but that is also gif?

Reply:
also includes heuristics are very weak, it would be preferable with a full list of supported values